tool/src/org/antlr/v4/automata/LexerATNFactory.java

```diff
@@ -546,52 +551,64 @@ private CharSetParseState applyPrevStateAndMoveToProperty(
 		return state;
 	}
 
-	private void applyPrevState(GrammarAST charSetAST, IntervalSet set, CharSetParseState state) {
+	private void applyPrevState(GrammarAST charSetAST, IntervalSet set, CharSetParseState state, boolean caseInsensitive) {
 		switch (state.mode) {
 			case NONE:
 			case ERROR:
 				break;
 			case PREV_CODE_POINT:
-				checkSetCollision(charSetAST, set, state.prevCodePoint);
-				set.add(state.prevCodePoint);
+				checkCharAndAddToSet(charSetAST, set, state.prevCodePoint, caseInsensitive);
 				break;
 			case PREV_PROPERTY:
 				set.addAll(state.prevProperty);
 				break;
 		}
 	}
 
-	protected void checkSetCollision(GrammarAST ast, IntervalSet set, int el) {
-		checkSetCollision(ast, set, el, el);
+	private void checkCharAndAddToSet(GrammarAST ast, IntervalSet set, int c, boolean caseInsensitive) {
+		checkRangeAndAddToSet(ast, set, c, c, caseInsensitive);
 	}
 
-	protected void checkSetCollision(GrammarAST ast, IntervalSet set, int a, int b) {
-		for (int i = a; i <= b; i++) {
-			if (set.contains(i)) {
-				String setText;
-				if (ast.getChildren() == null) {
-					setText = ast.getText();
-				}
-				else {
-					StringBuilder sb = new StringBuilder();
-					for (Object child : ast.getChildren()) {
-						if (child instanceof RangeAST) {
-							sb.append(((RangeAST) child).getChild(0).getText());
-							sb.append("..");
-							sb.append(((RangeAST) child).getChild(1).getText());
-						}
-						else {
-							sb.append(((GrammarAST)child).getText());
+	private void checkRangeAndAddToSet(GrammarAST ast, IntervalSet set, int a, int b, boolean caseInsensitive) {
+		if (caseInsensitive) {
+			int lowerA = Character.toLowerCase(a);
+			int upperA = Character.toUpperCase(a);
+			int lowerB = Character.toLowerCase(b);
+			int upperB = Character.toUpperCase(b);
+			if (lowerA == upperA && upperB == lowerB) {
+				checkRangeAndAddToSet(ast, set, a, b, false);
+			} else {
```

Comment:
hi. can we keep `}` and `else {` on diff lines? thanks :)

Reply:
IDEA reformats code block in such way. Does it matter a lot?

Reply:
At this point I think it's late in the game to change the style. That code is used the separate line style for 25 years. Some people have let that drift, but we might as well keep it to the original style as I notice it.